packages/front-end/components/Settings/Team/RoleSelector.tsx

```diff
@@ -12,9 +14,12 @@ export default function RoleSelector({
   setValue: (value: MemberRoleWithProjects) => void;
   showUpgradeModal: () => void;
 }) {
+  const id = useMemo(() => uniqid(), []);
```

Comment:
Why not just do this directly in `SingleRoleSelector`?  Is there any reason to pass in an id as a prop vs having the component generate random ids internally?

Reply:
I could do that in the `SingleRoleSelector` too if we prefer. The only thing would be readability—defining your own ID's is easier to read and debug than seeing a bunch of auto-generated ID's and class names. But it really doesn't matter, functionality-wise, since the goal is just to have different ID's, so if you'd like I can change it so that developers don't have to think about setting ID's when using this component.

Another option I tend to do when I build component libraries that include form input components like this is make the `id` optional and fall back to a generated ID, but I often separate out `<label>` and `<input />` components, so not sure this makes sense in this case.

Let me know what our preferred way would be and I can change it. I don't have strong feelings either way.

Reply:
Yeah, I'd probably change it so developers don't have to think about ids when using the component.

Reply:
Sounds good, change has been pushed up.